proposals/function-pointers.md

```diff
@@ -0,0 +1,245 @@
+# Function Pointers
+
+## Summary
+This proposal provides language constructs that expose IL opcodes that cannot currently be accessed efficiently,
+or at all, in C# today: `ldftn` and `calli`. These IL opcodes can be important in high performance code and developers
+need an effecient way to access them.
+
+## Motivation
+The motivations and background for this feature are described in the following issue (as is a 
+potential implementation of the feature): 
+
+https://github.com/dotnet/csharplang/issues/191
+
+This is an alternate design propsoal to [compiler intrinsics](https://github.com/dotnet/csharplang/blob/master/proposals/intrinsics.md)
+
+## Detailed Design 
+
+### funcptr
+The language will allow for the declaration of function pointers using the `funcptr` contextual keyword. The 
+declaration and usage of function pointers closely resemble that of `delegate`:
+
+``` csharp
+unsafe class Example {
+    delegate void DAction(int a);
+    funcptr void FAction(int a);
+
+    void Example(DAction d, FAction f) {
+        d(42);
+        f(42);
+    }
+}
+```
+
+These types are represented using the function pointer type as outlined in ECMA-335. This means invocation
```

Comment:
What is the named funcptr going to look like in the metadata? I know that ECMA-335 allows unnamed funcptrs, but I do not recall named funcptrs.

Reply:
Ok, I see that this is in the Open Issues.